core/src/Streamly/FileSystem/Path.hs

```diff
@@ -0,0 +1,212 @@
+-- |
+-- Module      : Streamly.FileSystem.Path
+-- Copyright   : (c) 2023 Composewell Technologies
+-- License     : BSD3
+-- Maintainer  : streamly@composewell.com
+-- Portability : GHC
+--
+-- = User Notes
```

Comment:
Do not add the docs in the exposed modules, leave it in the internal modules for now. We have to write it better when moving it. Currently it is just random notes, and developer notes should anyway remain in the internal modules.